python/cugraph/cugraph/structure/property_graph.py

```diff
@@ -1760,7 +1766,11 @@ def edge_props_to_graph(
             # Ensure a valid edge_weight_property can be used for applying
             # weights to the subgraph, and if a default_edge_weight was
             # specified, apply it to all NAs in the weight column.
-            if edge_weight_property in edge_prop_df.columns:
+            if edge_weight_property == self.type_col_name:
+                prop_col = edge_prop_df[self.type_col_name].cat.codes.astype("float32")
+                edge_prop_df["temp_type_col"] = prop_col
+                edge_weight_property = "temp_type_col"
```

Comment:
probably unlikely, but can you change these new names to have leading underscores to minimize the chance of a collision with existing user columns? Same for the MG version.

Reply:
done